polymetis/polymetis/python/polymetis/utils/continuous_grasper.py

```diff
@@ -2,49 +2,69 @@
 #
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
-import time
-
+import a0
+import json
 import numpy as np
+import os
+import pickle
+import platform
+import time
 import torch
-
 import torchcontrol as toco
+
+from polymetis import RobotInterface, GripperInterface
+from polymetis_pb2 import RobotState, GripperState
 from torchcontrol.transform import Rotation as R
 from torchcontrol.transform import Transformation as T
-from polymetis import RobotInterface, GripperInterface
+from typing import Dict, List
 
 
 DEFAULT_MAX_ITERS = 3
 
 # Sampling params
-GP_RANGE_UPPER = [0.7, 0.1, np.pi / 2]
-GP_RANGE_LOWER = [0.4, -0.1, -np.pi / 2]
+DEFAULT_GP_RANGE_UPPER = [0.7, 0.1, np.pi / 2]
+DEFAULT_GP_RANGE_LOWER = [0.4, -0.1, -np.pi / 2]
 
 # Grasp params
-REST_POSE = ([0.5, 0.0, 0.7], [1.0, 0.0, 0.0, 0.0])
-PREGRASP_HEIGHT = 0.55
-GRASP_HEIGHT = 0.25
-PLANNER_DT = 0.02
+DEFAULT_REST_POSE = ([0.5, 0.0, 0.7], [1.0, 0.0, 0.0, 0.0])
+DEFAULT_PREGRASP_HEIGHT = 0.55
+DEFAULT_GRASP_HEIGHT = 0.15
+DEFAULT_PLANNER_DT = 0.02
+DEFAULT_TIME_TO_GO_SECS = 2.0
 
 
 class ManipulatorSystem:
     def __init__(self, robot_kwargs={}, gripper_kwargs={}):
-        self.arm = RobotInterface(**robot_kwargs)
-        self.gripper = GripperInterface(**gripper_kwargs)
+        ip_address = robot_kwargs.get("ip_address", "localhost")
+        enforce_version = robot_kwargs.get("enforce_version", "True") == "True"
+        gripper_ip_address = gripper_kwargs.get("gripper_ip_address", "localhost")
+        self.arm = RobotInterface(
+            ip_address=ip_address, enforce_version=enforce_version
+        )
+        self.gripper = GripperInterface(ip_address=gripper_ip_address)
+        self.rest_pose = DEFAULT_REST_POSE
+        self.pregrasp_height = DEFAULT_PREGRASP_HEIGHT
+        self.grasp_height = DEFAULT_GRASP_HEIGHT
+        self.planner_dt = DEFAULT_PLANNER_DT
+        self.gp_range_upper = torch.Tensor(DEFAULT_GP_RANGE_UPPER)
+        self.gp_range_lower = torch.Tensor(DEFAULT_GP_RANGE_LOWER)
+        self.up_time_seconds = DEFAULT_TIME_TO_GO_SECS
+        self.down_time_seconds = DEFAULT_TIME_TO_GO_SECS
```

Comment:
Maybe we should just have this as input parameters with default values?

Reply:
I made these properties but given that this is meant to be used in a test setting i want the flexibility to control all of these.